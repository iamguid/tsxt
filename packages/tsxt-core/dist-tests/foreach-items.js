"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ctx = void 0;
var _foreachItems = _interopRequireDefault(require("./foreach-items.template"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const ctx = {
  items: [{
    name: "a",
    value: 1
  }, {
    name: "b",
    value: 2
  }, {
    name: "c",
    value: 3
  }]
};
exports.ctx = ctx;
function main() {
  const result = (0, _foreachItems.default)(ctx);
  console.log(result);
}
main();