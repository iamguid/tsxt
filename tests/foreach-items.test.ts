import template from "./foreach-items.template";
import { ctx } from "./foreach-items";

it("template generates correctly", () => {
  const result = template(ctx);
  expect(result).toMatchSnapshot();
});
