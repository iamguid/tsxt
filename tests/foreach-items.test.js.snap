// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template generates correctly 1`] = `
"item \\"a\\" has value \\"1\\";
item \\"b\\" has value \\"2\\";
item \\"c\\" has value \\"3\\";
"
`;
